### ISSUE TITLE
What are the overall aims of this template repo?

### ISSUE THREAD

**Issue by RobjS** [dxw/wordpress-template#35]
Even though this template repo is relatively new, we already have a large number of suggestions for changes we could make, some of which could potentially radically alter our technical approach to WordPress as a team. Reading through the issues we currently have open, it struck me that it was difficult to evaluate the priority of possible changes, and even to decide what we should work on at all.

I feel like a clearly stated set of aims - or, I suppose, user needs, with current and potential future members of the team as the users -  for the template repo might help with that. We would then have something to measure suggested changes against, allowing us to evaluate what we should work on and why.

Here are some initial suggestions for what the aims for this repo might be:

- Enabling us to avoid repetition of set-up tasks when building new sites, so we can start being productive more quickly
- Ensuring a consistent structure across our projects, so they're easier to pick up and start working on
- Enforcing technical standards across our work, so our code is more robust, maintainable, and testable
- Providing a central place for us to discuss and reach shared decisions on how we approach our work, with practical examples and outcomes to focus that discussion, so that everyone on the team understands how our projects work and why
- Helping new team members see what the basic structure of our projects is, so they can more easily understand more complex projects that have built upon it

I'd be interested to hear other suggestions. If we could come up with a handful of agreed aims, I feel like that would clarify our thinking when looking at future development on this repo.

**Comment by RobjS** [comment]
It might also help to think about the [development team principles](https://playbook.dxw.com/#development-team-principles) when we're formulating the aims. They are:

1. Solve real user needs
1. Own your code, but collaborate
1. Security is a user need
1. Use simple, conventional technology
1. Build for quality and consistency
1. Improve with each iteration
1. Communicate clearly and frequently
1. Be humble, supportive and open minded
1. Always learning. Always teaching
1. Do the smallest amount of good work
1. Work in the open

**Comment by mallorydxw** [comment]
I agree with all those suggestions. We could also document those purposes in a README somewhere.

**Comment by RobjS** [comment]
@anthonyomalley @calumryan @dxw-serena @mattymaxwell It would be good to get your opinions on this too.
